src/plugins/github/cacheId.js

```diff
@@ -9,14 +9,33 @@ export opaque type CacheId: string = string;
 
 /**
  * Derives the CacheId for a RepoId.
- * Returned CacheId's will be:
+ *
+ * Returned `CacheId`s will be:
  * - Deterministic
  * - Unique for this plugin
  * - Lowercase
  * - Safe to use for filenames
+ * - Distinct for semantically distinct inputs (input IDs that differ
+ *   only in case may map to the same output, because GitHub does not
+ *   permit collisions-modulo-case)
  */
 export function cacheIdForRepoId(repoId: RepoId): CacheId {
-  const repoString = repoIdToString(repoId);
-  const repoStringHex = Buffer.from(repoString).toString("hex");
-  return `github_${repoStringHex}`.toLowerCase();
+  // GitHub owner (user/organization) and repository names may be in
+  // mixed case, but GitHub prevents mixed-case collisions: e.g., if
+  // `foo` is a user then `Foo` cannot also be a user, and likewise for
+  // repositories. GitHub login names are DNS-safe (`[0-9A-Za-z-]` only)
+  // and so are filename-safe. Repository names have a slightly larger
+  // character set, including underscore, but are also filename-safe.
+  // Because login may not contain an underscore, it thus suffices to
+  // use an underscore as the delimiter. The resulting filename uses
+  // only `[0-9A-Za-z_.-]` and so is valid on all major filesystems.
+  const owner = repoId.owner.toLowerCase();
+  const name = repoId.name.toLowerCase();
+  if (owner.indexOf("_") !== -1) {
+    throw new Error(
+      "unexpected underscore in GitHub owner name would be ambiguous: " +
+        repoIdToString(repoId)
+    );
+  }
```

Comment:
Nit: This is technically redundant, as `RepoId` is an opaque type which already tests for the absence of underscores in it's factory function. In other instances we also rely on this to enforce certain properties that aren't available in Flow types.

For example: minimum array length.
https://github.com/sourcecred/sourcecred/blob/09739512ab14dc7a7a6825ffe069f671a02ad7d3/src/plugins/initiatives/initiative.js#L9-L19

Reply:
Yep; I commented as much on the test case (lines 69–71). It pays to be
extra defensive here because we’re constructing a cache key, so silent
collisions would be quite bad. This protects against the case where we
change the `RepoId` invariants and forget to update this caller. The
check is basically free since this code is rarely called, so I think
that it’s worth it.